tasks/image_size.js

```diff
@@ -7,53 +7,92 @@
-		var options = this.options({
-			configObject : ''
-		});
-
-		this.files.forEach(function (file) {
-			if ((!options.configObject || !options.configObject.length) && !file.dest) {
-				grunt.fail.warn('Neither options.configObject nor dest specified for this task.');
-				return;
-			}
-
-			var output = [];
-			file.src.forEach(function (src) {
-				if (grunt.file.exists(src) && grunt.file.isFile(src) && types.indexOf(path.extname(src).toLowerCase()) !== -1) {
-					var dimensions = sizer(src);
-					output.push({
-						name : src,
-						width : dimensions.width,
-						height : dimensions.height
-					});
-
-					grunt.verbose.writeln('dimensions of ' + src + ' width: ' + dimensions.width.toString() + ', height: ' + dimensions.height.toString());
-				}
-			});
-
-			if (options.configObject && options.configObject.length) {
-				grunt.config.set(options.configObject, output);
-			}
-
-			if (file.dest) {
-				grunt.file.write(file.dest, JSON.stringify(output));
-				grunt.log.writeln('File ' + chalk.cyan(file.dest) + ' created.');
-			}
-		});
-	});
+  grunt.registerMultiTask(
+    'image_size',
+    'Retrieve images size information',
+    function() {
+      var options = this.options({
+        processName: undefined,
+        processEntry: undefined,
+        processSizes: undefined,
+        configObject: undefined,
+        replacer: null,
+        space: 2
+      });
+
+      if (!this.files.length) {
+        return grunt.log.error('No files or `options.configObject` specified.');
+      }
+
+      var processedFiles = 0;
+
+      this.files.forEach(function(file) {
+        if (!file.src.length)
+          return grunt.log.error(
+            'No source files specified for ' + chalk.cyan(dest) + '.'
+          );
+
+        if (!file.dest) return grunt.log.error('No dest file specified.');
+
+        var sizes = [];
+
+        file.src.forEach(function(src) {
+          if (!grunt.file.isFile(src)) return;
+
+          var size = sizer(src);
+          var name = src;
+
+          if (typeof options.processName === 'function') {
+            name = options.processName.call(file, src, file);
+          }
+
+          var entry = { name: name, width: size.width, height: size.height };
+
+          if (typeof options.processEntry === 'function') {
+            entry = options.processEntry.call(file, entry, src, file);
+          }
+
+          sizes.push(entry);
+
+          grunt.verbose.writeln(
+            'Size of ' +
+              src +
+              ' width: ' +
+              size.width.toString() +
+              ', height: ' +
+              size.height.toString()
+          );
+
+          processedFiles++;
+        });
+
+        if (typeof options.processSizes === 'function') {
+          sizes = options.processSizes.call(file, sizes, file);
+        }
+
+        if (options.configObject && options.configObject.length) {
+          grunt.config.set(options.configObject, sizes);
+        }
+
+        grunt.file.write(
```

Comment:
you need to check, if file.dest is defined

Reply:
We're checking this earlier, to avoid running useless operations if task can't be accomplished anyway, here https://github.com/ArmorDarks/grunt-image-size/blob/7c445effedeeaa04ee98c97fd71acb5429825eeb/tasks/image_size.js#L36

Reply:
dest can be undefined and it is valid case if configObject defined (as in README: "outputs it to grunt Config Data and/or to dest file"), so we need to change condition in https://github.com/ArmorDarks/grunt-image-size/blob/7c445effedeeaa04ee98c97fd71acb5429825eeb/tasks/image_size.js#L36 and add it here

Reply:
Agree, fixed that.